caffe2/core/operator_schema.cc

```diff
@@ -352,24 +353,33 @@ static void SparseSegmentsFillerHelper(
       .Max(shapes[value_index].front() * 2)
       .Dist(FD_UNIFORM);
   (*fillers)[segment_index].SparseSegments(shapes[value_index].front() - 1);
+  if (weight_index < fillers->size()) {
+    (*fillers)[weight_index]
+        .Min(0)
+        .Max(shapes[weight_index].front() - 1)
+        .Dist(FD_UNIFORM);
+  }
 }
 
 // The helper is build sparse input with values, keys, and lengths; e.g.:
 // values  = [1, 2, 3, 2, 4, 6, 7, 3, 6]
 // keys    = [0, 1, 4, 0, 1, 2, 5, 1, 2]
+// weights = [1, 2, 3, 4, 5, 6, 7, 8, 9]
 //            \_____/  \________/  \__/
 // lengths =    [3,        4,       2]
-OpSchema& OpSchema::ValueKeyLengthInputFillers(
+OpSchema& OpSchema::WeightedValueKeyLengthInputFillers(
```

Comment:
If we separate the interfaces, `WeightedValueKeyLengthInputFillers` would be:
```lang=c++
WeightedValueKeyLengthInputFillers(...) {
  ValueKeyLengthInputFillers(...);
  // add assertion on weight_index if necessary
  (*fillers)[weight_index]
        .Min(0)
        .Max(shapes[weight_index].front() - 1)
        .Dist(FD_UNIFORM);
} 
```

Reply:
This doesn't work because ValueKeyLengthInputFillers sets a function. We need to go into the function creation phase.